cpp/src/stream_compaction/unique_count.cu

```diff
@@ -70,14 +71,26 @@ cudf::size_type unique_count(table_view const& keys,
                              null_equality nulls_equal,
                              rmm::cuda_stream_view stream)
 {
-  auto table_ptr = cudf::table_device_view::create(keys, stream);
-  row_equality_comparator comp(
-    nullate::DYNAMIC{cudf::has_nulls(keys)}, *table_ptr, *table_ptr, nulls_equal);
-  return thrust::count_if(
-    rmm::exec_policy(stream),
-    thrust::counting_iterator<cudf::size_type>(0),
-    thrust::counting_iterator<cudf::size_type>(keys.num_rows()),
-    [comp] __device__(cudf::size_type i) { return (i == 0 or not comp(i, i - 1)); });
+  auto const row_comp = cudf::experimental::row::equality::self_comparator(keys, stream);
+  // row_equality_comparator comp(
+  //   nullate::DYNAMIC{cudf::has_nulls(keys)}, *table_ptr, *table_ptr, nulls_equal);
+  if (cudf::detail::has_nested_columns(keys)) {
+    auto const comp =
+      row_comp.equal_to<true>(nullate::DYNAMIC{has_nested_nulls(keys)}, nulls_equal);
+    return thrust::count_if(
+      rmm::exec_policy(stream),
+      thrust::counting_iterator<cudf::size_type>(0),
+      thrust::counting_iterator<cudf::size_type>(keys.num_rows()),
+      [comp] __device__(cudf::size_type i) { return (i == 0 or not comp(i, i - 1)); });
+  } else {
+    auto const comp =
+      row_comp.equal_to<false>(nullate::DYNAMIC{has_nested_nulls(keys)}, nulls_equal);
+    return thrust::count_if(
+      rmm::exec_policy(stream),
+      thrust::counting_iterator<cudf::size_type>(0),
+      thrust::counting_iterator<cudf::size_type>(keys.num_rows()),
+      [comp] __device__(cudf::size_type i) { return (i == 0 or not comp(i, i - 1)); });
```

Comment:
Is it not worth doing the same `comparator_helper` trick here since the comparator is generated with a single simple line?

Reply:
Would have to create a functor for the device lambda then, since you can't have device lambdas inside host lambdas

Reply:
Yeah this sucks. We can't call device lambda inside host lambda :angry:.